source/auth/auth.rst

```diff
@@ -683,6 +683,259 @@ mechanism
+
+   ``mongodb://<access_key>:<secret_key>@mongodb.example.com/?authMechanism=MONGODB-IAM``
+
+Temporary IAM Credentials
+   If a username and password are not provided drivers MUST query the standard local AWS endpoint for temporary credentials. If 
+   temporary credentials cannot be obtained then drivers MUST fail authentication and raise an error. If an `AWS_SESSION_TOKEN` is
+   provided in addition to a username and password this is considered an `Assume Role <https://docs.aws.amazon.com/STS/latest/APIReference/API_AssumeRole.html>`_ request. An example URI for authentication with MONGODB-IAM using temporary IAM credentials from an EC2 instance
+   or ECS tasks is as follows:
+
+   ``mongodb://mongodb.example.com/?authMechanism=MONGODB-IAM``
+
+   An example using temporary IAM credentials as an Assume Role request is as follows:
+
+   ``mongodb://<access_key>:<secret_key>@mongodb.example.com/?authMechanism=MONGODB-IAM&authMechanismProperties=AWS_SESSION_TOKEN:<security_token>``
+
+.. note::
+   "session token" and "security token" are used interchangeably here and throughout the AWS documentation.
+
+   EC2 Instance
+      Calling the endpoint ``http://169.254.169.254/latest/meta-data/iam/security-credentials/`` from an EC2 instance 
+      will return the role attached to the EC2 instance in plaintext, if it exist:
+
+      ``role-name``
+
+      ``http://169.254.169.254/latest/meta-data/iam/security-credentials/<role-name>`` will return the following json response: 
+
+      .. code:: javascript
+
+            {
+                "Code": "Success",
+                "LastUpdated" : <date>,
+                "Type": "AWS-HMAC",
+                "AccessKeyId" : <access_key>,
+                "SecretAccessKey": <secret_access_key>,
+                "Token" : <security_token>,
+                "Expiration": <date>
+            }
+
+      .. note::
+
+      The IP address 169.254.169.254 is a link-local address and is valid only from the instance. 
+      See `Instance Metadata and User Data <https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2-instance-metadata.html>`_.
+
+      From this response, drivers will need to extract the ``AccessKeyId``, ``SecretAccessKey`` and the ``Token`` argument values. 
+      Drivers will use these values to construct the call to ``sts::getCallerIdentity`` as part of the Amazon Signature V4 algorithm. 
+      Drivers who cache the security temporary credentials must check the expiration date before using them. 
+      If the credentials expire, they must refetch them. 
+
+   ECS Tasks
+      ECS temporary credentials can be retrieved by calling a URL at ``http://169.254.170.2/$AWS_CONTAINER_CREDENTIALS_RELATIVE_URI`` 
+      where ``AWS_CONTAINER_CREDENTIALS_RELATIVE_URI`` is an environment variable set by the AWS ECS agent.
+
+      .. code:: javascript
+
+         {
+          "AccessKeyId": "ACCESS_KEY_ID",
```

Comment:
For clarity, consider using the same format here as above, e.g.

     "ACCESS_KEY_ID" => <access_key>